### PR TITLE
Azure SDK Intellij Plugin

### PR DESCRIPTION
**Pull request by srnagar**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This is the initial version of the Azure SDK IntelliJ plugin. This has the basic Maven rule checks. The purpose of this PR is to setup the plugin. The new inspections are not the final state and there will be some updates to this. This is also to keep the PR small and get early feedback.

Does this close any currently open issues?
------------------------------------------
None

Any relevant logs, screenshots, error output, etc.?
-------------------------------------
image

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [x] Tested

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/build.gradle [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+dependencies {
+    compileOnly project(path: ':azure-intellij-plugin-lib')
+    runtimeOnly project(path: ':azure-intellij-plugin-lib', configuration: 'instrumentedJar')
+    compileOnly project(path: ':azure-intellij-plugin-lib-java')
+    runtimeOnly project(path: ':azure-intellij-plugin-lib-java', configuration: 'instrumentedJar')
+    implementation 'com.microsoft.azure:azure-toolkit-common-lib'
+    implementation 'com.microsoft.azure:azure-toolkit-ide-common-lib'
+    aspect 'com.microsoft.azure:azure-toolkit-common-lib'
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/build.gradle:12]: this is only needed if you use our own telemetry framework.

File: PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle [hunk 2]
```diff
@@ -8,9 +8,9 @@ plugins {
 }
 
 processResources {
-    filesMatching('ApplicationInsights.xml') {
-        filter(ReplaceTokens, tokens: ["applicationinsights.key": project.property("applicationinsights.key")])
-    }
+//     filesMatching('ApplicationInsights.xml') {
+//         filter(ReplaceTokens, tokens: ["applicationinsights.key": project.property("applicationinsights.key")])
+//     }
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle]: don't commit this modification.

**Review by wangmingliang-ms — CHANGES_REQUESTED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/build.gradle (hunk 1 above)
Comment (srnagar) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/build.gradle:12]: I will keep this for now as I want to experiment with using your telemetry framework but with a different endpoint.

**Review by wangmingliang-ms — APPROVED** [review]
LGTM

**Comment by srnagar** [comment]
@wangmingliang-ms could you please help merge this PR? This is going into a feature branch. So, it should not impact the main branch or the upcoming release.

**Pull request merged**